sample/core.py

```diff
@@ -161,18 +171,18 @@ def get_categoryinfo(self):
                     continue
                 if page_id not in ci_list:
                     ci_list[page_id] = []
-                ci_list[page_id] = page_content[prop]
+                ci_list[page_id] += page_content[prop]
 
         self.payload.pop('cicontinue', None)
         self.payload['prop'] = None
         return ci_list
 
-    def get_duplicatefiles(self, dflimit="max", dfdir="ascending", dflocalonly=None):
+    def get_duplicatefiles(self, dflimit="max", dfdir="ascending", dflocalonly=False):
         prop = 'duplicatefiles'
         self.payload['prop'] = prop
         self.payload['dflimit'] = dflimit
         self.payload['dfdir'] = dfdir
-        if dflocalonly is not None:
+        if dflocalonly==True:
```

Comment:
I agree with the first point. Better would be to use `"Category:Infobox templates"`.

For the second point, say you instantiate a `WikiPage` object with many filenames. All will have the same pageid `-1`. How do you propose on keeping their results separate? Better would be another class for a `File`.